scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,157 @@
+      f.puts "#{changelog_body}\n\n"
+    }
+  end
+
+  desc "Increment versions"
+  private_lane :update_gradle_properties do |options|
+    version = options[:version].to_s
+    segments = version.match(version_regex).captures # version.split('.')
+
+    UI.message("Updating versionName in gradle.properties")
+
+    set_key_value(file: gradle_properties_path, key: 'VERSION_NAME', value: version)
+
+  end
+
+  # Calculate the Android version code. Leaving this here for posterity
+  # since it appears it's not necessary to increment versionCode for Android libraries.
+  desc "Calculate a version code based on new version segments"
+  private_lane :calculate_version_code do |options|
+    segments = options[:segments]
+    prerelease = segments.length() > 3 ?  0 : segments[-1].to_i
+    # Android version code max version = 2_100_000_000
+
+    # Derive the version code from version segements and pre-release number (if applicable)
+    # With this approach, you're "limited" to 999 minor versions per major and 999 patches per minor version.
+    # Limits: Max version for 1.x would be 1.999.999 with version code 199_999_900
+    segments[0].to_i * 100_000_000 + segments[1].to_i * 100_000 + segments[2].to_i * 100 + prerelease
+  end
+
+  desc "Create the release commit."
+  private_lane :release_commit do |options|
+    next_version = options[:version]
+    commit_message = "chore: release #{next_version} [skip ci]"
+    sh('git', 'commit', '-am', commit_message)
+  end
+
+  desc "Tag in git and push to GitHub"
+  private_lane :add_tag do |options|
+    next_version = options[:version]
+
+    add_git_tag(tag: next_version)
+    push_git_tags(tag: next_version)
+  end
+
+  desc "Publish release to maven using attributes from environment variables."
+  private_lane :publish_to_maven do |options|
+    gradle(
+      project_dir: gradle_project_root,
+      task: "uploadArchives",
+      print_command: false,
+      flags: '--stacktrace',
+      properties: {
+        "SONATYPE_NEXUS_USERNAME": ConfigValueReader.from_environment_variable('SONATYPE_NEXUS_USERNAME'),
+        "SONATYPE_NEXUS_PASSWORD": ConfigValueReader.from_environment_variable('SONATYPE_NEXUS_PASSWORD'),
+        "signing.inMemoryKey": ConfigValueReader.from_environment_variable('signing.inMemoryKey'),
+        "signing.password": ConfigValueReader.from_environment_variable('signing.password'),
+        "signing.keyId": ConfigValueReader.from_environment_variable('signing.keyId'),
+        "signing.secretKeyRingFile": ConfigValueReader.from_environment_variable('signing.secretKeyRingFile')
+      }
+    )
+  end
+
+  desc "Retrieve a parameter value from SSM."
+  private_lane :get_param_value do |options|
+    sh(
+      command: "aws ssm get-parameter --name #{options[:param_name]} --with-decryption --region us-east-1 --query Parameter.Value --output text",
+      log: false
+    )
+  end
+
+  desc "Cleanup all the build artifacts"
```

Comment:
When will this be called?

Reply:
I'm adding a call to it after the pull request is created.

Reply:
Hmm, what's the reasoning for that?  I would think the only time we need to call gradle clean is right before gradle build.   Or, even easier, I believe the [fastlane gradle command](https://docs.fastlane.tools/actions/gradle/) allows you to run multiple commands, like this:

```
gradle(
    tasks: ["clean", "build"]
)

Reply:
We always start from a clean state because a new runner is created every time the workflow is launched. The clean at the end is intended to get rid of any artifacts we wouldn't want to linger between the time the build ends and when Github actually destroys that runner. Probably not necessary since we don't have any sensitive information being handled by this workflow.

Reply:
Yeah...that makes sense. We're not really running a build here. I removed it.